dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/GeneralIT.java

```diff
@@ -105,4 +145,426 @@ public void testRemoteLaunchWDLNoFile() {
+    }
+
+    /**
+     * Tests that tool2JSON works for entries on Dockstore
+     */
+    @Test
+    public void testTool2JSONWDL() {
+        Client.main(new String[] { "--config", ResourceHelpers.resourceFilePath("config_file2.txt"), "tool", "publish", "--entry",
+            "quay.io/dockstoretestuser2/quayandgithubwdl" });
+        // need to publish before converting
+        Client.main(
+            new String[] { "--config", ResourceHelpers.resourceFilePath("config_file2.txt"), "tool", "convert", "entry2json", "--entry",
+                "quay.io/dockstoretestuser2/quayandgithubwdl", "--descriptor", "wdl", "--script" });
+        // TODO: Test that output is the expected WDL file
+        Assert.assertTrue(systemOutRule.getLog().contains("\"test.hello.name\": \"String\""));
+    }
+
+    @Test
+    public void registerUnregisterAndCopy() {
+        Client.main(new String[] { "--config", ResourceHelpers.resourceFilePath("config_file2.txt"), "tool", "publish", "--entry",
+            "quay.io/dockstoretestuser2/quayandgithubwdl" });
+
+        boolean published = testingPostgres.runSelectStatement(
+            "select ispublished from tool where registry = '" + Registry.QUAY_IO.toString()
+                + "' and namespace = 'dockstoretestuser2' and name = 'quayandgithubwdl';", boolean.class);
+        assertTrue("tool not published", published);
+
+        Client.main(new String[] { "--config", ResourceHelpers.resourceFilePath("config_file2.txt"), "tool", "publish", "--entry",
+            "quay.io/dockstoretestuser2/quayandgithubwdl", "--entryname", "foo" });
+
+        long count = testingPostgres.runSelectStatement("select count(*) from tool where registry = '" + Registry.QUAY_IO.toString()
+            + "' and namespace = 'dockstoretestuser2' and name = 'quayandgithubwdl';", long.class);
+        assertEquals("should be two after republishing", 2, count);
+
+        Client.main(
+            new String[] { "--config", ResourceHelpers.resourceFilePath("config_file2.txt"), "tool", "publish", "--unpub", "--entry",
+                "quay.io/dockstoretestuser2/quayandgithubwdl" });
+
+        published = testingPostgres.runSelectStatement("select ispublished from tool where registry = '" + Registry.QUAY_IO.toString()
+            + "' and namespace = 'dockstoretestuser2' and name = 'quayandgithubwdl' and toolname IS NULL;", boolean.class);
+        assertFalse(published);
+    }
+
+    /**
+     * Tests that WDL2JSON works for local file
+     */
+    @Test
+    public void testWDL2JSON() {
+        File sourceFile = new File(ResourceHelpers.resourceFilePath("wdl.wdl"));
+        Client.main(new String[] { "--config", ResourceHelpers.resourceFilePath("config_file2.txt"), "tool", "convert", "wdl2json", "--wdl",
+            sourceFile.getAbsolutePath(), "--script" });
+        // TODO: Test that output is the expected WDL file
+    }
+
+    @Test
+    @Category(ToilCompatibleTest.class)
```

Comment:
Do we need this category still? Not sure what it is doing.

Reply:
I am unsure, can ask Denis when he is back.